src/Tools/dotnet-monitor/CollectionRules/Configuration/CollectionRulePostConfigureNamedOptions.cs

```diff
@@ -76,117 +64,79 @@ private bool ResolveActionList(CollectionRuleOptions ruleOptions, string name)
-            if (section.Exists() && !string.IsNullOrEmpty(section.Value))
+            IConfigurationSection section = ruleSection.GetSection(nameof(CollectionRuleOptions.Limits));
+
+            if (!SectionHasValue(section))
             {
-                if (!TryGetTemplate(_templateOptions.CollectionRuleLimits, section.Value, out CollectionRuleLimitsOptions limitsOptions))
-                {
-                    return false;
-                }
+                TryGetTemplate(ruleOptions, _templateOptions.CollectionRuleLimits, section.Value, out CollectionRuleLimitsOptions limitsOptions);
 
                 ruleOptions.Limits = limitsOptions;
             }
-
-            return true;
-        }
-
-        private bool TryGetTemplate<T>(IDictionary<string, T> templatesOptions, string templateKey, out T templatesValue)
-        {
-            if (templatesOptions.TryGetValue(templateKey, out templatesValue))
-            {
-                return true;
-            }
-
-            _logger.LogError(Strings.ErrorMessage_TemplateNotFound, templateKey);
-            return false;
         }
 
-        private void BindActionSettings(IConfigurationSection actionSection, CollectionRuleActionOptions actionOptions)
+        private void TryGetTemplate<T>(CollectionRuleOptions ruleOptions, IDictionary<string, T> templatesOptions, string templateKey, out T templatesValue) where T : new()
         {
-            if (null != actionOptions &&
-                _actionOperations.TryCreateOptions(actionOptions.Type, out object actionSettings))
+            if (!templatesOptions.TryGetValue(templateKey, out templatesValue))
             {
-                IConfigurationSection settingsSection = actionSection.GetSection(
-                    nameof(CollectionRuleActionOptions.Settings));
-
-                settingsSection.Bind(actionSettings);
-                actionOptions.Settings = actionSettings;
+                templatesValue = new();
+                ruleOptions.ErrorList.Add(new ValidationResult(string.Format(CultureInfo.CurrentCulture, Strings.ErrorMessage_TemplateNotFound, templateKey)));
             }
         }
 
-        private void BindTriggerSettings(IConfigurationSection triggerSection, CollectionRuleTriggerOptions triggerOptions)
+        private bool SectionHasValue(IConfigurationSection section)
         {
-            if (null != triggerOptions &&
-                _triggerOperations.TryCreateOptions(triggerOptions.Type, out object triggerSettings))
-            {
-                IConfigurationSection settingsSection = triggerSection.GetSection(nameof(CollectionRuleTriggerOptions.Settings));
-
-                settingsSection.Bind(triggerSettings);
-
-                triggerOptions.Settings = triggerSettings;
-            }
+            // If the section has a value, the value is the name of a template.
```

Comment:
Isn't it the opposite though? "If the section does not have a value, then the key is the name of the template"

Reply:
Actually, I think the comment is right, but I just have two inversions cancelling each other out. It should be `!string.IsNullOrEmpty(section.Value)`, and then the caller should be `SectionHasValue(section)`